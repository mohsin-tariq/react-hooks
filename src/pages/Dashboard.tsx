import { useHistory } from 'react-router-dom'

const Dashboard = (): JSX.Element => {
  const history = useHistory()

  return (
    <>
      <h1>Dashboard Page</h1>
      <button
        type="button"
        className="btn"
        cy-data="go-back-button"
        onClick={() => history.push('/')}
      >
        Go back
      </button>
    </>
  )
}

export default Dashboard
