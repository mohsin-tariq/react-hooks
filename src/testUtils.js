
/** eslint-disable **/
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'

const initialState = {
  zone: {
    data: [
      {
        countryCode: 'AD',
        countryName: 'Andorra',
        zoneName: 'Europe/Andorra',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
      {
        countryCode: 'AL',
        countryName: 'Albania',
        zoneName: 'Europe/Albania',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
    ],
  },
}

const mockStore = configureMockStore()

function render(
  ui,
  { store = mockStore(initialState), ...renderOptions } = {}
) {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
