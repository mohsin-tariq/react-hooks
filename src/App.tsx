import { BrowserRouter, Switch, Route } from 'react-router-dom'
import Navbar from './components/Navbar'
import Buttons from './pages/Buttons'
import Dashboard from './pages/Dashboard'

import Home from './pages/Home'

const App = (): JSX.Element => {
  return (
    <BrowserRouter>
      <Navbar />
      <div className="container">
        <Switch>
          <Route path="/dashboard" component={Dashboard} />
          <Route path="/buttons" component={Buttons} />
          <Route path="/" component={Home} />
        </Switch>
      </div>
    </BrowserRouter>
  )
}

export default App
