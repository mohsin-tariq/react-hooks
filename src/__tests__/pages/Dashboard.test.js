import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import App from '../../App'
import Dashboard from '../../pages/Dashboard'
import { render, screen, fireEvent } from '../../testUtils'

describe('Routing', () => {
  test('route to Dashboard and go back to Home Page', () => {
    const history = createMemoryHistory()
    render(
      <Router history={history}>
        <App />
      </Router>
    )
    fireEvent.click(screen.getByText('Dashboard'))
    expect(screen.getByText('Dashboard Page')).toBeInTheDocument()

    fireEvent.click(screen.getByText('Go back'))
    expect(screen.getByText('Time Zone Component')).toBeInTheDocument()
  })

  test('route to HomePage', () => {
    const history = createMemoryHistory()
    const { asFragment } = render(
      <Router history={history}>
        <Dashboard />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })
  
})
