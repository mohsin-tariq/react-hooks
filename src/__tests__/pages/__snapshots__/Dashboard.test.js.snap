// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routing route to HomePage 1`] = `
<DocumentFragment>
  <h1>
    Dashboard Page
  </h1>
  <button
    class="btn"
    cy-data="go-back-button"
    type="button"
  >
    Go back
  </button>
</DocumentFragment>
`;
